pkg/types/config.go

```diff
@@ -38,25 +39,18 @@ type SecretConfig struct {
 }
 
 // BuildURL constructs a URL which can be used to identify a HTTP based secret provider
-func (c SecretConfig) BuildURL(path string) (spURL string, err error) {
+func (c SecretConfig) BuildURL(path string) string {
 	// Make sure there is not a trailing slash
 	if strings.HasSuffix(path, "/") {
 		path = path[:len(path)-1]
 	}
 
-	rawUrl := fmt.Sprintf("%s://%s:%v%s", c.Protocol, c.Host, c.Port, path)
-
-	u, err := url.Parse(rawUrl)
-	if err != nil {
-		return "", err
-	}
-
-	return u.String(), nil
+	return fmt.Sprintf("%s://%s:%v%s", c.Protocol, c.Host, c.Port, path)
```

Comment:
where the validation of url logic goes?

Reply:
Not needed, we are building it be hand. The requestor displays the URL if there is an error.

Reply:
I disagree, the Protocol string can be input from user and could be anything: why not validate?

Reply:
Ok

Reply:
More robust validation added back.